src/functions.php

```diff
@@ -63,9 +63,14 @@ function uploadImages(array $vcards, $config, $configPhonebook, callable $callba
     $configImagepath = rtrim($configImagepath, '/') . '/';  // ensure one slash at end
 
     // Prepare FTP connection
-    $ftpserver = $config['url'];
-    $ftpserver = str_replace("http://", "", $ftpserver);    // config.example.php has http:// which breaks FTP connect
-    $ftp_conn = ftp_connect($ftpserver);
+    $ftpserver = (parse_url($config['url'], PHP_URL_HOST) ? parse_url($config['url'], PHP_URL_HOST) : $config['url']);
+    $useFTPS = true;                                                                                                                                                                                           
+    if (isset($config['usePlainFTP']) && $config['usePlainFTP']) { $useFTPS =  false; }                                                                                                                        
+    if (!function_exists("ftp_ssl_connect")) {
+        error_log(PHP_EOL."WARNING: Falling back to plain text ftp.");
+        $useFTPS =  false;
+    }                                                                                                                                                                                                          
+    $ftp_conn  = ($useFTPS ? ftp_ssl_connect($ftpserver) : ftp_connect($ftpserver));
```

Comment:
It would probably be a little simpler to just call the function by name:

    $connectFunc = (isset($config['usePlainFTP']) && $config['usePlainFTP']) ? 'ftp_connect' : ftp_ssl_connect';

and simplify the entire paragraph.